src/upgradeExecutorPrepareRemoveExecutorTransactionRequest.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Address,
+  PublicClient,
+  encodeFunctionData,
+  PrepareTransactionRequestReturnType,
+} from 'viem';
+import { upgradeExecutor } from './contracts';
+import {
+  UPGRADE_EXECUTOR_ROLE_EXECUTOR,
+  upgradeExecutorEncodeFunctionData,
+} from './upgradeExecutorEncodeFunctionData';
+
+/**
+ * Type for the params of the upgradeExecutorPrepareRemoveExecutorTransactionRequest function
+ */
+export type UpgradeExecutorPrepareRemoveExecutorTransactionRequestParams = {
+  account: Address;
+  upgradeExecutorAddress: Address;
+  executorAccountAddress: Address;
+  publicClient: PublicClient;
+};
+
+/**
+ * Prepares a transaction to revoke the executor role from an account
+ *
+ * - Example: [Add new executor account to UpgradeExecutor](https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/examples/upgrade-executor-add-account/index.ts)
+ *
+ * @param {UpgradeExecutorPrepareRemoveExecutorTransactionRequestParams} upgradeExecutorPrepareRemoveExecutorTransactionRequestParams {@link UpgradeExecutorPrepareRemoveExecutorTransactionRequestParams}
+ * @param {Address} upgradeExecutorPrepareRemoveExecutorTransactionRequestParams.account - Address of the account to be revoked the executor role
+ * @param {Address} upgradeExecutorPrepareRemoveExecutorTransactionRequestParams.upgradeExecutorAddress - Address of the UpgradeExecutor contract
+ * @param {Address} upgradeExecutorPrepareRemoveExecutorTransactionRequestParams.executorAccountAddress - Address of the account that already has the executor role (will send the transaction)
+ * @param {PublicClient} upgradeExecutorPrepareRemoveExecutorTransactionRequestParams.publicClient - The chain Viem Public Client
+ *
+ * @returns Promise<{@link PrepareTransactionRequestReturnType}> - the transaction request
+ *
+ * @example
+ * const removeExecutorTransactionRequest = await upgradeExecutorPrepareRemoveExecutorTransactionRequest({
+ *   account: accountAddress,
+ *   upgradeExecutorAddress: coreContracts.upgradeExecutor,
+ *   executorAccountAddress: deployer.address,
+ *   publicClient,
+ * });
+ */
+export async function upgradeExecutorPrepareRemoveExecutorTransactionRequest({
+  account,
+  upgradeExecutorAddress,
+  executorAccountAddress,
+  publicClient,
+}: UpgradeExecutorPrepareRemoveExecutorTransactionRequestParams) {
+  const chainId = publicClient.chain!.id;
```

Comment:
We can call `validateParentChain` instead here

Reply:
Can we have any case where `publicClient.chain` doesn't exist? If so, I think we should throw error rather than using non-null assertion. Maybe we can create a `validatePublicClient` util

Reply:
Following the parentChain example, I added a validateChain helper function: https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/b13f0599df92d467c3b1410dbf9adf31870033bd/src/utils/validateChain.ts#L11